File: backend/internal/dto/lesson.go
package dto

type LessonSummaryDTO struct {
	ID         uint   `json:"id"`
	Number     int    `json:"number"`
	Title      string `json:"title"`
	CoverImage string `json:"coverImage"`
	UnitCount  int    `json:"unitCount"`
}

type LessonDetailDTO struct {
	ID         uint             `json:"id"`
	Number     int              `json:"number"`
	Title      string           `json:"title"`
	CoverImage string           `json:"coverImage"`
	Objectives []ObjectiveDTO   `json:"objectives"`
	Units      []UnitSummaryDTO `json:"units"`
}

type ObjectiveDTO struct {
	ID            uint   `json:"id"`
	UnitCode      string `json:"unitCode"`
	Skill         string `json:"skill"`
	Grammar       string `json:"grammar"`
	Vocabulary    string `json:"vocabulary"`
	Pronunciation string `json:"pronunciation"`
	Writing       string `json:"writing"`
}

File: web/language_learning/modules/lesson/types/lesson.ts
import type { Unit } from "@/types/dto/unit";

/**
 * Main lesson entity.
 *
 * Mirrors backend:
 * ParsedLesson / models.Lesson
 */
export interface Lesson {
  id: string;

  number: number;

  title: string;

  description?: string;

  coverImage?: string;

  units: Unit[];

  objectives: LessonObjective[];

  createdAt?: string;

  updatedAt?: string;
}


/**
 * Learning objectives extracted from textbook metadata.
 *
 * Example:
 *
 * Grammar:
 *   Present simple
 *
 * Vocabulary:
 *   Greetings
 */
export interface LessonObjective {
  id?: string;

  unitCode?: string;

  skill?: string;

  grammar?: string;

  vocabulary?: string;

  pronunciation?: string;

  writing?: string;
}


/**
 * Lightweight lesson card model.
 *
 * Used in:
 * - dashboard
 * - lesson list
 * - search results
 */
export interface LessonSummary {
  id: string;

  number: number;

  title: string;

  coverImage?: string;

  unitCount: number;

  completedUnits: number;

  progress: number;
}


/**
 * User lesson progress.
 */
export interface LessonProgress {
  lessonId: string;

  completedUnits: number;

  totalUnits: number;

  completedExercises: number;

  totalExercises: number;

  percentage: number;

  lastVisitedAt?: string;
}


/**
 * API response for lesson list.
 */
export interface LessonListResponse {
  items: LessonSummary[];

  total: number;

  page: number;

  pageSize: number;
}


/**
 * API response for single lesson.
 */
export interface LessonResponse {
  data: Lesson;
}